modules/post/windows/gather/enum_chocolatey_applications.rb

```diff
@@ -0,0 +1,97 @@
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+
+class MetasploitModule < Msf::Post
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Windows Gather Installed Application Within Chocolatey Enumeration',
+        'Description' => ' This module will enumerate all installed applications on a Windows system with chocolatey installed ',
+        'License' => MSF_LICENSE,
+        'Author' => ['Nick Cottrell <ncottrellweb@gmail.com>'],
+        'Platform' => ['win'],
+        'Privileged' => false,
+        'SessionTypes' => %w[meterpreter shell],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => []
+        }
+      )
+    )
+    register_advanced_options(
+      [
+        OptString.new('ChocoPath', [false, 'The path to the chocolaty executable if its not on default path', 'choco.exe']),
+      ]
+    )
+  end
+
+  def chocopath
+    # cmd_exec('where.exe', 'choco.exe') unless chocolatey?
+
+    if chocolatey?
+      datastore['ChocoPath']
+    end
+  rescue StandardError
+    cmd_exec('where.exe', 'choco.exe')
+  end
+
+  def chocolatey?
+    cmd_exec(datastore['ChocoPath'], '-v') =~ /\d+\.\d+\.\d+/m
+  end
+
+  def run
+    # checking that session is meterpreter and session has powershell
+    return 0 unless chocopath
```

Comment:
Instead of returning zero, I'd suggest using the `fail_with()` method. In this case, the failure could look something like:

```ruby
fail_with(Failure::NotFound, 'Chocolatey path not found') unless chocopath
```